invoices.go

```diff
@@ -0,0 +1,186 @@
+	Duration         string    `json:"duration"`
+	DurationUnit     string    `json:"duration_unit"`
+	StartTime        time.Time `json:"start_time"`
+	EndTime          time.Time `json:"end_time"`
+	ProjectName      string    `json:"project_name"`
+}
+
+// InvoiceList contains a paginated list of all of a customer's invoices.
+// The InvoicePreview is the month-to-date usage generated by DigitalOcean.
+type InvoiceList struct {
+	Invoices       []InvoiceListItem `json:"invoices"`
+	InvoicePreview InvoiceListItem   `json:"invoice_preview"`
+	Links          *Links            `json:"links"`
+	Meta           *Meta             `json:"meta"`
+}
+
+// InvoiceListItem contains a small list of information about a customer's invoice.
+// More information can be found in the Invoice or InvoiceSummary
+type InvoiceListItem struct {
+	InvoiceUUID   string    `json:"invoice_uuid"`
+	Amount        string    `json:"amount"`
+	InvoicePeriod string    `json:"invoice_period"`
+	UpdatedAt     time.Time `json:"updated_at"`
+}
+
+// InvoiceSummary contains metadata and summarized usage for an invoice generated by DigitalOcean
+type InvoiceSummary struct {
+	InvoiceUUID           string                  `json:"invoice_uuid"`
+	BillingPeriod         string                  `json:"billing_period"`
+	Amount                string                  `json:"amount"`
+	UserName              string                  `json:"user_name"`
+	UserBillingAddress    Address                 `json:"user_billing_address"`
+	UserCompany           string                  `json:"user_company"`
+	UserEmail             string                  `json:"user_email"`
+	ProductCharges        InvoiceSummaryBreakdown `json:"product_charges"`
+	Overages              InvoiceSummaryBreakdown `json:"overages"`
+	Taxes                 InvoiceSummaryBreakdown `json:"taxes"`
+	CreditsAndAdjustments InvoiceSummaryBreakdown `json:"credits_and_adjustments"`
+}
+
+// Address represents the billing address of a customer
+type Address struct {
+	AddressLine1    string    `json:"address_line1"`
+	AddressLine2    string    `json:"address_line2"`
+	City            string    `json:"city"`
+	Region          string    `json:"region"`
+	PostalCode      string    `json:"postal_code"`
+	CountryISO2Code string    `json:"country_iso2_code"`
+	CreatedAt       time.Time `json:"created_at"`
+	UpdatedAt       time.Time `json:"updated_at"`
+}
+
+// InvoiceSummaryBreakdown is a grouped set of InvoiceItems from an invoice
+type InvoiceSummaryBreakdown struct {
+	Name   string                        `json:"name"`
+	Amount string                        `json:"amount"`
+	Items  []InvoiceSummaryBreakdownItem `json:"items"`
+}
+
+// InvoiceSummaryBreakdownItem further breaks down the InvoiceSummary by product
+type InvoiceSummaryBreakdownItem struct {
+	Name   string `json:"name"`
+	Amount string `json:"amount"`
+	Count  string `json:"count"`
+}
+
+func (i Invoice) String() string {
+	return Stringify(i)
```

Comment:
What's the benefit of hiding this in another method?

Is `i.String()` functionally better than `Stringify(i)` or is it just better visually?

Reply:
there's no functional improvement, I just see it used in other instances.  It doesn't seem to be a hard requirement, I included it for consistency